### PR TITLE
Reduce docker image size.

### PR DESCRIPTION
**Pull request by qingling128**

### CODE REVIEW

**Comment by qingling128** [comment]
`debian:stretch-slim` is a moving target. But numbered images (e.g. `9.5-slim`) are static.

**Comment by qingling128** [comment]
Tried a few different things. So far the smallest image ended up with `444MB` with the following Dockerfile:

```
FROM debian:7.11-slim

# TODO: This may be a moving target, figure out how to pin.
RUN apt-get update \
    && apt-get install -y \
        curl \
        gnupg2 \
        lsb-release \
        build-essential \
        procps \
        systemd \
    # Install google-fluentd at version 1.5.35-1.
    && curl -sS https://dl.google.com/cloudagents/install-logging-agent.sh | REPO_SUFFIX=20180810-1 DO_NOT_INSTALL_CATCH_ALL_CONFIG=true bash \
    && apt-get remove -y \
        curl \
        lsb-release \
        build-essential \
    && rm -rf /var/lib/apt/lists/* /tmp/* \
    && apt-get autoremove -y \
    && find /var/log -name "*.log" -type f -delete \
    && sed -i "s/num_threads 8/num_threads 8\n  detect_json true\n  # Enable metadata agent lookups.\n  enable_metadata_agent true\n  metadata_agent_url \"http:\/\/local-metadata-agent.stackdriver.com:8000\"/" "/etc/google-fluentd/google-fluentd.conf"

ENV LD_PRELOAD=/opt/google-fluentd/embedded/lib/libjemalloc.so
ADD entrypoint.sh /entrypoint.sh

ENTRYPOINT ["/entrypoint.sh"]
CMD ["/usr/sbin/google-fluentd"]
```

Note that when the order of commands got shifted as below:
```
FROM debian:7.11-slim

# TODO: This may be a moving target, figure out how to pin.
RUN apt-get update \
    && apt-get install -y \
        curl \
        gnupg2 \
        lsb-release \
        build-essential \
        procps \
        systemd \
    # Install google-fluentd at version 1.5.35-1.
    && curl -sS https://dl.google.com/cloudagents/install-logging-agent.sh | REPO_SUFFIX=20180810-1 DO_NOT_INSTALL_CATCH_ALL_CONFIG=true bash \
    && sed -i "s/num_threads 8/num_threads 8\n  detect_json true\n  # Enable metadata agent lookups.\n  enable_metadata_agent true\n  metadata_agent_url \"http:\/\/local-metadata-agent.stackdriver.com:8000\"/" "/etc/google-fluentd/google-fluentd.conf" \
    && rm -rf /var/lib/apt/lists/* /tmp/* \
    && apt-get autoremove -y \
    && find /var/log -name "*.log" -type f -delete \
    && apt-get remove -y \
        curl \
        lsb-release \
        build-essential
```

The image size increased to `549MB`.

The prior image size was `559MB`:
```
$ docker image ls gcr.io/stackdriver-agents/stackdriver-logging-agent:0.4-1.5.35-1-k8s
REPOSITORY                                            TAG                 IMAGE ID            CREATED             SIZE
gcr.io/stackdriver-agents/stackdriver-logging-agent   0.4-1.5.35-1-k8s    3915b01e1314        3 weeks ago         559MB
```

**Comment by qingling128** [comment]
With doc removed, we are down to `424MB`.

**Comment by qingling128** [comment]
With `--no-install-recommends`, we are down to `389MB`.

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
-    && apt-get install -y \
+    && apt-get install -y --no-install-recommends \
         curl \
         gnupg2 \
         lsb-release \
         build-essential \
         procps \
         systemd \
+        ca-certificates \
+        adduser \
```
Comment (igorpeshansky) [inline docker/Dockerfile]: For the future, should either of these two be a dependency of `google-fluentd`? Then they would be caught by `apt-get autoremove`.

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
-RUN curl -sS https://dl.google.com/cloudagents/install-logging-agent.sh | REPO_SUFFIX=20180810-1 DO_NOT_INSTALL_CATCH_ALL_CONFIG=true bash \
-    && rm -rf /var/lib/apt/lists/* /tmp/* \
-    && apt-get autoremove \
+    && apt-get autoremove -y \
     && find /var/log -name "*.log" -type f -delete \
+    && rm -rf /usr/share/doc /usr/share/man /opt/google-fluentd/embedded/share/doc /opt/google-fluentd/embedded/lib/ruby/gems/2.4.0/doc \
     && sed -i "s/num_threads 8/num_threads 8\n  detect_json true\n  # Enable metadata agent lookups.\n  enable_metadata_agent true\n  metadata_agent_url \"http:\/\/local-metadata-agent.stackdriver.com:8000\"/" "/etc/google-fluentd/google-fluentd.conf"
 
 ENV LD_PRELOAD=/opt/google-fluentd/embedded/lib/libjemalloc.so
-
```
Comment (igorpeshansky) [inline docker/Dockerfile]: Let's keep this blank line.

Comment (qingling128) [reply docker/Dockerfile]: Done

File: docker/Dockerfile (hunk 1 above)
Comment (qingling128) [reply docker/Dockerfile]: Created a ticket for this.

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,22 +1,26 @@
-FROM debian:stretch-20180426
+FROM debian:7.11-slim
 
 # TODO: This may be a moving target, figure out how to pin.
```
Comment (igorpeshansky) [inline docker/Dockerfile]: Just realized that this comment now applies to the `FROM` line as well...

**Pull request merged**